### PR TITLE
Updating sources in runtime styling section

### PR DESCRIPTION
**Pull request by langsmith**
This pr updates the runtime styling doc and mainly adds more info about the different sources that available to Android devs. This page needed updating.

### CODE REVIEW

**Comment by langsmith** [comment]
cc @tobrun @LukasPaczos in case I'm missing any sources or I'm incorrect about my additions in this pr. Wasn't sure whether `UnknownSource` should be mentioned or whether that's just for native/internal use...

File: src/pages/map-sdk/overview/runtime-styling.md [hunk 1]
```diff
@@ -121,13 +119,66 @@ mapboxMap.addSource(geoJsonSource);
...
+CustomGeometrySource source = new CustomGeometrySource(ID_GRID_SOURCE, GeometryTileProvider);
+mapboxMap.addSource(source);
+```
+
+One example of `CustomGeometrySource` usage is to create a black grid on top of the map. This example's code can be found in [the `GridSourceActivity` of the Maps SDK for Android test application](https://github.com/mapbox/mapbox-gl-native/blob/4498917a3b9dbf6cc9728da01f479a027f27f902/platform/android/MapboxGLAndroidSDKTestApp/src/main/java/com/mapbox/mapboxsdk/testapp/activity/style/GridSourceActivity.java).
+
+
+### Raster Dem
+
+`RasterDemSource` currently only supports Mapbox Terrain RGB (mapbox://mapbox.terrain-rgb). The Mapbox terrain tileset is for adding hill terrain to any Mapbox map. Runtime styling can also be used to change the hillshade appearance.
```
Comment (LukasPaczos) [inline src/pages/map-sdk/overview/runtime-styling.md]: With https://github.com/mapbox/mapbox-gl-native/pull/11339 we are also supporting mapzen terrarium encoded sources.

**Comment by langsmith** [comment]
@LukasPaczos , added a blurb about Mapzen . Anything else?

**Comment by LukasPaczos** [comment]
Looks good to me @langsmith :+1:

File: src/pages/map-sdk/overview/runtime-styling.md [hunk 2]
```diff
@@ -121,13 +119,68 @@ mapboxMap.addSource(geoJsonSource);
 
 A benefit of having your data inside a GeoJSON source is that you can update, remove, or add additional features inside the source at any time, providing a solution to animating data in your map through the Runtime Styling API. For example, using an Android [ValueAnimator](https://developer.android.com/reference/android/animation/ValueAnimator.html), you can move a feature by updating its coordinates within the GeoJSON data.
 
-<!-- NOTE link to GoJSON plugin -->
+### Image
+
+`ImageSource` allows for a georeferenced raster image to be shown on top of the map. The georeferenced image scales and rotates as the user zooms and rotates the map. The geographic location of the raster image content, supplied with `LatLngQuad`, can be non-axis aligned.
+
+```
```
Comment (colleenmcginnis) [inline src/pages/map-sdk/overview/runtime-styling.md]: Use ` ```java ` for syntax highlighting.

File: src/pages/map-sdk/overview/runtime-styling.md [hunk 3]
```diff
@@ -121,13 +119,68 @@ mapboxMap.addSource(geoJsonSource);
...
+mapboxMap.addSource(new ImageSource(ID_IMAGE_SOURCE, quad, R.drawable));
+
+// Add layer
+RasterLayer layer = new RasterLayer(ID_IMAGE_LAYER, ID_IMAGE_SOURCE);
+mapboxMap.addLayer(layer);
+```    
+
+The `setImage()` method is a convenient way to update the `ImageSource`'s image by passing in a drawable.
+
+```
```
Comment (colleenmcginnis) [inline src/pages/map-sdk/overview/runtime-styling.md]: Use ` ```java ` for syntax highlighting.

File: src/pages/map-sdk/overview/runtime-styling.md [hunk 4]
```diff
@@ -121,13 +119,68 @@ mapboxMap.addSource(geoJsonSource);
...
+mapboxMap.getSource(ID_IMAGE_SOURCE)).setImage(R.drawable.image_to_use)
+```
+
+### Custom Geometry
+
+There might be a situation when you want to draw a shape that doesn't fit the standard `Point`, `LineString`, `Polygon`, `MultiPoint`, `MultiLineString`, and `MultiPolygon` GeoJSON geometries. A `CustomGeometrySource` can help you achieve this. 
+
+```
```
Comment (colleenmcginnis) [inline src/pages/map-sdk/overview/runtime-styling.md]: Use ` ```java ` for syntax highlighting.

File: src/pages/map-sdk/overview/runtime-styling.md [hunk 5]
```diff
@@ -121,13 +119,68 @@ mapboxMap.addSource(geoJsonSource);
...
+One example of `CustomGeometrySource` usage is to create a black grid on top of the map. This example's code can be found in [the `GridSourceActivity` of the Maps SDK for Android test application](https://github.com/mapbox/mapbox-gl-native/blob/4498917a3b9dbf6cc9728da01f479a027f27f902/platform/android/MapboxGLAndroidSDKTestApp/src/main/java/com/mapbox/mapboxsdk/testapp/activity/style/GridSourceActivity.java).
+
+
+### Raster Dem
+
+`RasterDemSource` currently supports Mapbox Terrain RGB (mapbox://mapbox.terrain-rgb) and [Mapzen Terrarium](https://mapzen.com/documentation/terrain-tiles/formats/#terrarium) tile formats.
+
+The Mapbox terrain tileset is for adding hill terrain to any Mapbox map. Runtime styling can also be used to change the hillshade appearance.
+
+```
```
Comment (colleenmcginnis) [inline src/pages/map-sdk/overview/runtime-styling.md]: Use ` ```java ` for syntax highlighting.

File: src/pages/map-sdk/overview/runtime-styling.md [hunk 6]
```diff
@@ -121,13 +119,68 @@ mapboxMap.addSource(geoJsonSource);
...
+mapboxMap.addLayer(layer);
+```    
+
+The `setImage()` method is a convenient way to update the `ImageSource`'s image by passing in a drawable.
+
+```
+mapboxMap.getSource(ID_IMAGE_SOURCE)).setImage(R.drawable.image_to_use)
+```
+
+### Custom Geometry
```
Comment (colleenmcginnis) [inline src/pages/map-sdk/overview/runtime-styling.md]: Use sentence case : "Custom geometry"

File: src/pages/map-sdk/overview/runtime-styling.md [hunk 7]
```diff
@@ -121,13 +119,68 @@ mapboxMap.addSource(geoJsonSource);
...
+
+```
+CustomGeometrySource source = new CustomGeometrySource(ID_GRID_SOURCE, GeometryTileProvider);
+mapboxMap.addSource(source);
+```
+
+One example of `CustomGeometrySource` usage is to create a black grid on top of the map. This example's code can be found in [the `GridSourceActivity` of the Maps SDK for Android test application](https://github.com/mapbox/mapbox-gl-native/blob/4498917a3b9dbf6cc9728da01f479a027f27f902/platform/android/MapboxGLAndroidSDKTestApp/src/main/java/com/mapbox/mapboxsdk/testapp/activity/style/GridSourceActivity.java).
+
+
+### Raster Dem
```
Comment (colleenmcginnis) [inline src/pages/map-sdk/overview/runtime-styling.md]: Should this be "Raster DEM"?

File: src/pages/map-sdk/overview/runtime-styling.md (hunk 4 above)
Comment (langsmith) [reply src/pages/map-sdk/overview/runtime-styling.md]: Fixed

File: src/pages/map-sdk/overview/runtime-styling.md (hunk 5 above)
Comment (langsmith) [reply src/pages/map-sdk/overview/runtime-styling.md]: Fixed

File: src/pages/map-sdk/overview/runtime-styling.md (hunk 7 above)
Comment (langsmith) [reply src/pages/map-sdk/overview/runtime-styling.md]: Yep, fixed

File: src/pages/map-sdk/overview/runtime-styling.md (hunk 6 above)
Comment (langsmith) [reply src/pages/map-sdk/overview/runtime-styling.md]: Fixed

File: src/pages/map-sdk/overview/runtime-styling.md (hunk 2 above)
Comment (langsmith) [reply src/pages/map-sdk/overview/runtime-styling.md]: Fixed

File: src/pages/map-sdk/overview/runtime-styling.md (hunk 3 above)
Comment (langsmith) [reply src/pages/map-sdk/overview/runtime-styling.md]: Fixed

**Comment by langsmith** [comment]
Anything else @colleenmcginnis

**Pull request merged**